app/src/main/java/org/fossasia/openevent/general/MainActivity.kt

```diff
@@ -88,10 +91,12 @@ class MainActivity : AppCompatActivity() {
     override fun onBackPressed() {
         val currentFragment = this.supportFragmentManager.findFragmentById(R.id.frameContainer)
         val rootFragment = this.supportFragmentManager.findFragmentById(R.id.rootLayout)
-        if (rootFragment is EventDetailsFragment)
-            super.onBackPressed()
-        else
-            when (currentFragment) {
+        when (rootFragment) {
+            is EventDetailsFragment,
+            is TicketsFragment,
+            is AttendeeFragment,
+            is OrderDetailsFragment -> super.onBackPressed()
+            else -> when (currentFragment) {
```

Comment:
Both when conditions can be written separately. Let mentors review it.

Reply:
@iamareebjamal how about this the rootFragment is null when we are in the main page so

      if (rootFragment == null) {
                 when (currentFragment) {
                    is SearchFragment,
                    is FavoriteFragment,
                    is OrdersUnderUserFragment,
                    is ProfileFragment -> {
                        loadFragment(supportFragmentManager, EventsFragment(), frameContainer.id)
                        navigation.selectedItemId = navigation_events
                    }
                    is EventsFragment -> finish()
                    else -> super.onBackPressed()
                }
            }
        else super.onBackPressed()

Reply:
Seems better but reverse the condition and use early return

Reply:
@iamareebjamal like this 


    if (rootFragment != null) {
                super.onBackPressed()
                return
            }
        when (currentFragment) {
            is SearchFragment,
            is FavoriteFragment,
            is OrdersUnderUserFragment,
            is ProfileFragment -> {
                loadFragment(supportFragmentManager, EventsFragment(), frameContainer.id)
                navigation.selectedItemId = navigation_events
            }
            is EventsFragment -> finish()
            else -> super.onBackPressed()
        }
    }

Reply:
Yes